### DISCUSSION TITLE
Dynamically adding plugins/components?

### DISCUSSION THREAD

**Discussion by addlistener** [GrapesJS/grapesjs#5996] [Q&A] [answered]
Let's say I have 
```
const ComponentA = (editor) => editor.DomComponents.addType(...)
const ComponentB = (editor) => editor.DomComponents.addType(...)
...
```

I checked the docs https://grapesjs.com/docs/modules/Plugins.html#basic-plugin you can only add plugins when calling `init`. Is there a way we can do this dynamically? `editor.addPlugin(...)`.

The use case will be supporting dynamic components.

**Comment by artf** [comment, marked as answer]
The plugin is usually a simple function, so you could call it by passing the current editor instance and the options you need.
```js
import someGjsPlugin from 'someGjsPlugin';
// ...
const editor = grapesjs.init({...});
// ...
someGjsPlugin(editor, { ...plugin options.. });
```

**Reply by clonefunnels** [reply]
When I try this it breaks the block manager set in the javascript file.

**Reply by lexoyo** [reply]
Hi
You'll probably need a reproductible demo, on codepen for example

**Reply by lexoyo** [reply]
Please don't double post
i will try to answer here: https://github.com/GrapesJS/grapesjs/discussions/6383
